module/platform/Webots/src/Webots.cpp

```diff
@@ -404,11 +405,16 @@ namespace module::platform {
                             // Decode the protocol buffer and emit it as a message
                             char* payload = reinterpret_cast<char*>(buffer.data()) + sizeof(length);
 
-                            translate_and_emit_sensor(
-                                NUClear::util::serialise::Serialise<SensorMeasurements>::deserialise(payload, length));
-
-                            // Delete the packet we just read ready to read the next one
-                            buffer.erase(buffer.begin(), std::next(buffer.begin(), sizeof(length) + length));
+                            try {
```

Comment:
If you're going to do a try/catch like this i'd be careful you could end up in a bad state forever.

Maybe only service the watchdog down here on success

Reply:
Actually wait, this try/catch is really bad, you're in a loop here and you only erase from the buffer if you successfully deserialise the protobuf.
So if you fail, you'll go back around this loop, over and over again

Reply:
Yea, this was a hack ibadded to stop the program crashing and to grab some extra info. Should get rid of the try/catch